cmd/minikube/cmd/config/disable.go

```diff
@@ -38,11 +40,24 @@ var addonsDisableCmd = &cobra.Command{
 		if addon == "heapster" {
 			exit.Message(reason.AddonUnsupported, "The heapster addon is depreciated. please try to disable metrics-server instead")
 		}
-		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+		cc, err := config.Load(ClusterFlagValue())
 		if err != nil {
-			exit.Error(reason.InternalAddonDisable, "disable failed", err)
+			exit.Error(reason.InternalAddonDisable, "loading profile", err)
+		}
+		validAddon, ok := assets.Addons[addon]
+		if ok {
+			if validAddon.IsEnabled(cc) {
+				err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+				if err != nil {
+					exit.Error(reason.InternalAddonDisable, "disable failed", err)
+				}
+			} else {
+				out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
```

Comment:
I don't think this should be in an `else`. Before after `addons.SetAndSave` was called, if it was successful it would output `The '{{.minikube_addon}}' addon is disabled`. But now it's only outputting the message if the addon is already disabled.

Reply:
I think you are correct